### ISSUE TITLE
Failing Test_newTCPConn_ErrorsWhenTCPBindingSucceedsButConnectionFails

### ISSUE THREAD

**Issue by sacurio** [coyim/coyim#731]
The test is failing in Windows with the following output:

```
tcp_test.go:219:
    c.Check(addr, Equals, "xmpp.olabini.se:5222")
... obtained string = "olabini.se:5222"
... expected string = "xmpp.olabini.se:5222"

tcp_test.go:226:
    c.Check(err, Equals, errors.ErrConnectionFailed)
... obtained *errors.errorString = &errors.errorString{s:"failed to find a TCP address for XMPP server"} ("failed to find a TCP address for XMPP server")
... expected *errors.errorString = &errors.errorString{s:"could not connect to XMPP server"} ("could not connect to XMPP server")
... Difference:
...     s: "failed to find a TCP address for XMPP server" != "could not connect to XMPP server"

```

**Comment by olabiniV2** [comment]
Is this still failing?

**Comment by peterpalau** [comment]
This is not failing anymore and we don't have a clear idea of why. Our assumption is that it failed due to a bad configuration of the Windows virtual machine. Anyway, it's not failing anymore.

**Comment by olabiniV2** [comment]
Hmm OK. I'll close it for now, and set up Windows CI.

**Comment by olabiniV2** [comment]
Nope, this now fails in the CI.

**Comment by peterpalau** [comment]
Doing some research in the code about this, I found the following function:

https://github.com/coyim/coyim/blob/ffc6c132428e0785288afdf2d522a805b1c96ff8/xmpp/tcp.go#L86-L95

If no `xmppAddrs` are passed to the function (an empty slice), the function returns the error: `errors.ErrConnectionFailed ` which is the one we expect in the test.

But when we call this function, the returned error is discarded and an error obtained previously is returned instead.

https://github.com/coyim/coyim/blob/ffc6c132428e0785288afdf2d522a805b1c96ff8/xmpp/tcp.go#L78-L81

The problem here is, that in Windows no `addrs` are found but also, no error is returned by the `resolveSRVWithProxy` .

https://github.com/coyim/coyim/blob/ffc6c132428e0785288afdf2d522a805b1c96ff8/xmpp/tcp.go#L67-L68

In summary, `srvLookupAndFallback` returns a `ErrTCPBindingFailed` when no address are found and, the function `connectToFirstAvailable` returns a `ErrConnectionFailed` error in the same situation but, we discard that error when we call it from the `srvLookupAndFallback` function.

**Comment by olabiniV2** [comment]
Is this fixed now, since the Windows CI works? Or is it disabled or something?

**Comment by peterpalau** [comment]
> Is this fixed now, since the Windows CI works? Or is it disabled or something?

It's fixed. All the work related to this issue was referenced.

**Comment by olabiniV2** [comment]
Reviewed, OK
